File: csharp/reverse-string/ReverseStringTests.cs
using Xunit;

public class ReverseStringTests
{
    [Fact]
    public void An_empty_string()
    {
        Assert.Equal("", ReverseString.Reverse(""));
    }

    [Fact]
    public void A_word()
    {
        Assert.Equal("tobor", ReverseString.Reverse("robot"));
    }

    [Fact]
    public void A_capitalized_word()
    {
        Assert.Equal("nemaR", ReverseString.Reverse("Ramen"));
    }

    [Fact]
    public void A_sentence_with_punctuation()
    {
        Assert.Equal("!yrgnuh m'I", ReverseString.Reverse("I'm hungry!"));
    }

    [Fact]
    public void A_palindrome()
    {
        Assert.Equal("racecar", ReverseString.Reverse("racecar"));
    }

    [Fact]
    public void An_even_sized_word()
    {
        Assert.Equal("reward", ReverseString.Reverse("drawer"));
    }
}

File: fsharp/reverse-string/ReverseStringTests.fs
module ReverseStringTests

open FsUnit.Xunit
open Xunit

open ReverseString

[<Fact>]
let ``An empty string`` () =
    reverse "" |> should equal ""

[<Fact>]
let ``A word`` () =
    reverse "robot" |> should equal "tobor"

[<Fact>]
let ``A capitalized word`` () =
    reverse "Ramen" |> should equal "nemaR"

[<Fact>]
let ``A sentence with punctuation`` () =
    reverse "I'm hungry!" |> should equal "!yrgnuh m'I"

[<Fact>]
let ``A palindrome`` () =
    reverse "racecar" |> should equal "racecar"

[<Fact>]
let ``An even-sized word`` () =
    reverse "drawer" |> should equal "reward"

